core/client/hdfs/src/main/java/alluxio/hadoop/HadoopConfigurationUtils.java

```diff
@@ -36,10 +37,11 @@ private HadoopConfigurationUtils() {} // Prevent instantiation.
   /**
    * Merges Hadoop {@link org.apache.hadoop.conf.Configuration} into the Alluxio configuration.
    *
+   * @param uri the uri that may contain zookeeper configuration
    * @param source the {@link org.apache.hadoop.conf.Configuration} to merge
    * @param alluxioConfiguration the Alluxio configuration to merge to
    */
-  public static void mergeHadoopConfiguration(org.apache.hadoop.conf.Configuration source,
+  public static void mergeHadoopConfiguration(URI uri, org.apache.hadoop.conf.Configuration source,
```

Comment:
from an API perspective it's weird to have a URI as an argument for a util method about merging hadoop configuration. When I see just `mergeHadoopConfig(hadoopConfig, alluxioConfig)` it's pretty easy to guess what it does, but it isn't easy to guess what `mergeConfig(uri, hadoopConfig, alluxioConfig)` does.

Reply:
Removed the getting zookeeper configuration outside this method. The mergeHadoopConfiguration now will only merge hadoop and Alluxio configuration.